xc/device.ml

```diff
@@ -2682,6 +2698,50 @@ module Dm = struct
 
   (* the following functions depend on the functions above that use the qemu backend Q *)
 
+  let start_varstored ~xs ?(nvram=[]) ?(restore=false) task domid =
+    let path = !Xc_resources.varstored in
+    let name = "varstored" in
+    let arg x = [ "--arg"; x ] in
+    let argf fmt = ksprintf (fun x -> arg x) fmt in
+    let onf cond = ksprintf (fun x -> if cond then arg x else []) in
+    let vm_uuid = Xenops_helpers.uuid_of_domid ~xs domid |> Uuidm.to_string in
+    let reset_on_boot = match List.assoc "EFI-variables-on-boot" nvram with
+      | "persist" -> false
+      | "reset" -> true
+      | _ -> failwith "TODO: bad var"
+      | exception Not_found -> false
+    in
+    let efivars_init_arg = match List.assoc "EFI-variables" nvram with
+      | efivars ->
+        let efivars_init_path = efivars_init_path domid in
+        Unixext.write_string_to_file efivars_init_path efivars;
+        argf "init:%s" efivars_init_path
+      | exception Not_found -> []
```

Comment:
According to the design, if the EFI variables are not present in the db or they are empty or the VM is being resumed, then the init argument should not be given.